### PR TITLE
Tests for `Cache-Control` header parsing

### PR DESCRIPTION
**Pull request by marki1an**

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsFetchingSpec.groovy [hunk 1]
```diff
@@ -1575,6 +1575,38 @@ class PriceFloorsFetchingSpec extends PriceFloorsBaseSpec {
         invalidSkipRate << [MIN_SKIP_RATE - 1, MAX_SKIP_RATE + 1]
     }
 
+    def "PBS should reject fetch when response header is invalid"() {
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsFetchingSpec.groovy:1578]: Wrong check. PBS should not reject the auction when the header is invalid.

File: src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsFetchingSpec.groovy [hunk 2]
```diff
@@ -1575,6 +1575,38 @@ class PriceFloorsFetchingSpec extends PriceFloorsBaseSpec {
...
+
+        then: "PBS log should contain error"
+        def logs = floorsPbsService.getLogsByTime(startTime)
+        def floorsLogs = getLogsByText(logs, basicFetchUrl)
+        assert floorsLogs.size() == 1
+        assert floorsLogs[0].contains("Can't parse Cache Control header '${header.get(headerKey) as String}', "
+                + "fetch.url: '$basicFetchUrl$accountId'")
+
+        and: "Floors validation failure cannot reject the entire auction"
+        assert !response.seatbid?.isEmpty()
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsFetchingSpec.groovy:1607]: You should add a check that rule fetch has statred.

File: src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsFetchingSpec.groovy [hunk 3]
```diff
@@ -1575,6 +1575,38 @@ class PriceFloorsFetchingSpec extends PriceFloorsBaseSpec {
...
+
+        and: "Account with enabled fetch, fetch.url in the DB"
+        def accountId = bidRequest.site.publisher.id
+        def account = getAccountWithEnabledFetch(accountId)
+        accountDao.save(account)
+
+        and: "Set Floors Provider response with header"
+        def floorsResponse = PriceFloorData.priceFloorData
+        def headerKey = "Cache-Control"
+        def header = [(headerKey): "no-cache, no-store, max-age=0, must-revalidate"]
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsFetchingSpec.groovy:1593]: Please create a test to make sure that PBS honors this parameter by redownloading rules after max age.

**Review by Net-burst — CHANGES_REQUESTED** [review]

**Comment by marki1an** [comment]
Corresponding ticket: https://github.com/prebid/prebid-server-java/pull/1858

**Pull request closed without merging**